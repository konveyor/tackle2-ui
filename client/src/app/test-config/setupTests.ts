import "@testing-library/jest-dom";
import { server } from "@mocks/server";

const mockInitialized = false;

jest.mock("@react-keycloak/web", () => {
  const originalModule = jest.requireActual("@react-keycloak/web");
  return {
    ...originalModule,
    useKeycloak: () => [mockInitialized],
  };
});

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "localhost:3000/example/path",
  }),
}));

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

// Uncomment the following to see request logging in the console:

// server.events.on("request:start", (req) => {
//   console.log(`Handling a request to ${req.url.href}`);
// });

// server.events.on("request:match", (req) => {
//   console.log(`Request to ${req.url.href} was matched with a handler`);
// });

// server.events.on("request:unhandled", (req) => {
//   console.warn(`Request to ${req.url.href} was not handled`);
// });
