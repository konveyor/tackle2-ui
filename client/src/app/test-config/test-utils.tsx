import React, { FC, ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { Application, Archetype, Assessment } from "@app/api/models";
import { RenderHookOptions, renderHook } from "@testing-library/react-hooks";

import { createContext, useContext } from "react";

const QueryClientContext = createContext<QueryClient | undefined>(undefined);

const AllTheProviders: FC<{
  children: React.ReactNode;
  queryClient?: QueryClient;
}> = ({ children, queryClient }) => {
  const internalQueryClient =
    queryClient ||
    new QueryClient({
      defaultOptions: {
        queries: {
          retry: false,
          cacheTime: 1000,
        },
      },
    });
  return (
    <QueryClientProvider client={internalQueryClient}>
      {children}
    </QueryClientProvider>
  );
};

export const useQueryClientContext = () => {
  const context = useContext(QueryClientContext);
  if (context === undefined) {
    throw new Error(
      "useQueryClientContext must be used within a QueryClientContext.Provider"
    );
  }
  return context;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

const customRenderHook = <TProps, TResult>(
  callback: (props: TProps) => TResult,
  options?: Omit<RenderHookOptions<TProps>, "wrapper"> & {
    queryClient?: QueryClient;
  }
) => {
  const { queryClient, ...rest } = options || {};
  const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => (
    <AllTheProviders queryClient={queryClient}>{children}</AllTheProviders>
  );

  return renderHook(callback, { wrapper: Wrapper as React.FC, ...rest });
};

export * from "@testing-library/react";

export { customRender as render };
export { customRenderHook as renderHook };

export const createMockAssessment = (
  overrides: Partial<Assessment> = {}
): Assessment => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    required: true,
    ...overrides,
  } as Assessment;
};

export const createMockApplication = (overrides: Partial<Application> = {}) => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    ...overrides,
  } as Application;
};

export const createMockArchetype = (overrides: Partial<Archetype> = {}) => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    ...overrides,
  } as Archetype;
};
