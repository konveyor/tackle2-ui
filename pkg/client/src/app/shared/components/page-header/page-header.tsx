import React from "react";
import {
  Stack,
  StackItem,
  Split,
  SplitItem,
  TextContent,
  Text,
} from "@patternfly/react-core";
import { BreadCrumbPath } from "../breadcrumb-path";
import { MenuActions } from "../menu-actions";
import { HorizontalNav } from "../horizontal-nav/horizontal-nav";

export interface PageHeaderProps {
  title: string;
  description?: React.ReactNode;
  breadcrumbs: { title: string; path: string | (() => void) }[];
  btnActions?: React.ReactNode;
  menuActions: { label: string; callback: () => void }[];
  navItems?: { title: string; path: string }[];
}

export const PageHeader: React.FC<PageHeaderProps> = ({
  title,
  description,
  breadcrumbs,
  btnActions,
  menuActions,
  navItems,
}) => {
  return (
    <Stack hasGutter>
      <StackItem>
        {breadcrumbs.length > 0 && <BreadCrumbPath breadcrumbs={breadcrumbs} />}
      </StackItem>
      <StackItem>
        <Split>
          <SplitItem isFilled>
            <TextContent>
              <Text component="h1">{title}</Text>
              {description}
            </TextContent>
          </SplitItem>
          {btnActions && <SplitItem>{btnActions}</SplitItem>}
          {menuActions.length > 0 && (
            <SplitItem>
              <MenuActions actions={menuActions} />
            </SplitItem>
          )}
        </Split>
      </StackItem>
      {navItems && (
        <StackItem>
          <HorizontalNav navItems={navItems} />
        </StackItem>
      )}
    </Stack>
  );
};
