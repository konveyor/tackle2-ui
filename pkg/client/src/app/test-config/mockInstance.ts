import MockAdapter from "axios-mock-adapter";
import axios from "axios";

export default new MockAdapter(axios);
