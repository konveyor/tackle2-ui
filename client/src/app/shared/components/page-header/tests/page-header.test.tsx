import React from "react";
import { PageHeader } from "../page-header";
import { Button } from "@patternfly/react-core";
import { render } from "@app/test-config/test-utils";
import { BrowserRouter as Router } from "react-router-dom";

describe("PageHeader", () => {
  it("Renders without crashing", () => {
    const wrapper = render(
      <Router>
        <PageHeader
          title="mycompany"
          breadcrumbs={[
            {
              title: "Companies",
              path: "/companies",
            },
            {
              title: "Company details",
              path: "/companies/1",
            },
          ]}
          btnActions={<Button>send email</Button>}
          menuActions={[
            {
              label: "Edit",
              callback: jest.fn(),
            },
            {
              label: "Delete",
              callback: jest.fn(),
            },
          ]}
          navItems={[
            {
              title: "Overview",
              path: "/companies/1/overview",
            },
            {
              title: "YAML",
              path: "/companies/1/yaml",
            },
            {
              title: "SUNAT",
              path: "/companies/1/sunat",
            },
          ]}
        />
      </Router>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
