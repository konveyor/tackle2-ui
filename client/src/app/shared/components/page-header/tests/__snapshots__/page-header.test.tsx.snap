// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader Renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pf-l-stack pf-m-gutter"
      >
        <div
          class="pf-l-stack__item"
        >
          <nav
            aria-label="Breadcrumb"
            class="pf-c-breadcrumb"
            data-ouia-component-id="OUIA-Generated-Breadcrumb-1"
            data-ouia-component-type="PF4/Breadcrumb"
            data-ouia-safe="true"
          >
            <ol
              class="pf-c-breadcrumb__list"
            >
              <li
                class="pf-c-breadcrumb__item"
              >
                <a
                  class="pf-c-breadcrumb__link"
                  href="/companies"
                >
                  Companies
                </a>
              </li>
              <li
                class="pf-c-breadcrumb__item"
              >
                <span
                  class="pf-c-breadcrumb__item-divider"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </span>
                Company details
              </li>
            </ol>
          </nav>
        </div>
        <div
          class="pf-l-stack__item"
        >
          <div
            class="pf-l-split"
          >
            <div
              class="pf-l-split__item pf-m-fill"
            >
              <div
                class="pf-c-content"
              >
                <h1
                  class=""
                  data-ouia-component-id="OUIA-Generated-Text-1"
                  data-ouia-component-type="PF4/Text"
                  data-ouia-safe="true"
                  data-pf-content="true"
                >
                  mycompany
                </h1>
              </div>
            </div>
            <div
              class="pf-l-split__item"
            >
              <button
                aria-disabled="false"
                class="pf-c-button pf-m-primary"
                data-ouia-component-id="OUIA-Generated-Button-primary-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                send email
              </button>
            </div>
            <div
              class="pf-l-split__item"
            >
              <div
                class="pf-c-dropdown"
                data-ouia-component-id="OUIA-Generated-Dropdown-1"
                data-ouia-component-type="PF4/Dropdown"
                data-ouia-safe="true"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="pf-c-dropdown__toggle"
                  data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                  data-ouia-component-type="PF4/DropdownToggle"
                  data-ouia-safe="true"
                  id="pf-dropdown-toggle-id-0"
                  type="button"
                >
                  <span
                    class="pf-c-dropdown__toggle-text"
                  >
                    Actions
                  </span>
                  <span
                    class="pf-c-dropdown__toggle-icon"
                  >
                    <test-file-stub />
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="pf-l-stack__item"
        >
          <div
            class="pf-c-tabs"
          >
            <ul
              class="pf-c-tabs__list"
            >
              <a
                class="pf-c-tabs__item"
                href="/companies/1/overview"
              >
                <li
                  class="pf-c-tabs__item"
                >
                  <button
                    class="pf-c-tabs__link"
                  >
                    <span
                      class="pf-c-tabs__item-text"
                    >
                      Overview
                    </span>
                  </button>
                </li>
              </a>
              <a
                class="pf-c-tabs__item"
                href="/companies/1/yaml"
              >
                <li
                  class="pf-c-tabs__item"
                >
                  <button
                    class="pf-c-tabs__link"
                  >
                    <span
                      class="pf-c-tabs__item-text"
                    >
                      YAML
                    </span>
                  </button>
                </li>
              </a>
              <a
                class="pf-c-tabs__item"
                href="/companies/1/sunat"
              >
                <li
                  class="pf-c-tabs__item"
                >
                  <button
                    class="pf-c-tabs__link"
                  >
                    <span
                      class="pf-c-tabs__item-text"
                    >
                      SUNAT
                    </span>
                  </button>
                </li>
              </a>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <nav
          aria-label="Breadcrumb"
          class="pf-c-breadcrumb"
          data-ouia-component-id="OUIA-Generated-Breadcrumb-1"
          data-ouia-component-type="PF4/Breadcrumb"
          data-ouia-safe="true"
        >
          <ol
            class="pf-c-breadcrumb__list"
          >
            <li
              class="pf-c-breadcrumb__item"
            >
              <a
                class="pf-c-breadcrumb__link"
                href="/companies"
              >
                Companies
              </a>
            </li>
            <li
              class="pf-c-breadcrumb__item"
            >
              <span
                class="pf-c-breadcrumb__item-divider"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                  />
                </svg>
              </span>
              Company details
            </li>
          </ol>
        </nav>
      </div>
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-l-split"
        >
          <div
            class="pf-l-split__item pf-m-fill"
          >
            <div
              class="pf-c-content"
            >
              <h1
                class=""
                data-ouia-component-id="OUIA-Generated-Text-1"
                data-ouia-component-type="PF4/Text"
                data-ouia-safe="true"
                data-pf-content="true"
              >
                mycompany
              </h1>
            </div>
          </div>
          <div
            class="pf-l-split__item"
          >
            <button
              aria-disabled="false"
              class="pf-c-button pf-m-primary"
              data-ouia-component-id="OUIA-Generated-Button-primary-1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              send email
            </button>
          </div>
          <div
            class="pf-l-split__item"
          >
            <div
              class="pf-c-dropdown"
              data-ouia-component-id="OUIA-Generated-Dropdown-1"
              data-ouia-component-type="PF4/Dropdown"
              data-ouia-safe="true"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="pf-c-dropdown__toggle"
                data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                data-ouia-component-type="PF4/DropdownToggle"
                data-ouia-safe="true"
                id="pf-dropdown-toggle-id-0"
                type="button"
              >
                <span
                  class="pf-c-dropdown__toggle-text"
                >
                  Actions
                </span>
                <span
                  class="pf-c-dropdown__toggle-icon"
                >
                  <test-file-stub />
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-tabs"
        >
          <ul
            class="pf-c-tabs__list"
          >
            <a
              class="pf-c-tabs__item"
              href="/companies/1/overview"
            >
              <li
                class="pf-c-tabs__item"
              >
                <button
                  class="pf-c-tabs__link"
                >
                  <span
                    class="pf-c-tabs__item-text"
                  >
                    Overview
                  </span>
                </button>
              </li>
            </a>
            <a
              class="pf-c-tabs__item"
              href="/companies/1/yaml"
            >
              <li
                class="pf-c-tabs__item"
              >
                <button
                  class="pf-c-tabs__link"
                >
                  <span
                    class="pf-c-tabs__item-text"
                  >
                    YAML
                  </span>
                </button>
              </li>
            </a>
            <a
              class="pf-c-tabs__item"
              href="/companies/1/sunat"
            >
              <li
                class="pf-c-tabs__item"
              >
                <button
                  class="pf-c-tabs__link"
                >
                  <span
                    class="pf-c-tabs__item-text"
                  >
                    SUNAT
                  </span>
                </button>
              </li>
            </a>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
